app/src/main/java/org/schabi/newpipe/util/SecondaryStreamHelper.java

```diff
@@ -29,33 +31,37 @@ public SecondaryStreamHelper(final StreamSizeWrapper<T> streams, final T selecte
      * @param videoStream  desired video ONLY stream
      * @return selected audio stream or null if a candidate was not found
      */
+    @Nullable
     public static AudioStream getAudioStreamFor(@NonNull final List<AudioStream> audioStreams,
                                                 @NonNull final VideoStream videoStream) {
-        switch (videoStream.getFormat()) {
-            case WEBM:
-            case MPEG_4:// ¿is mpeg-4 DASH?
-                break;
-            default:
-                return null;
-        }
+        final MediaFormat mediaFormat = videoStream.getFormat();
+        if (mediaFormat != null) {
+            switch (mediaFormat) {
+                case WEBM:
+                case MPEG_4:// ¿is mpeg-4 DASH?
+                    break;
+                default:
+                    return null;
+            }
 
-        final boolean m4v = videoStream.getFormat() == MediaFormat.MPEG_4;
+            final boolean m4v = mediaFormat == MediaFormat.MPEG_4;
 
-        for (final AudioStream audio : audioStreams) {
-            if (audio.getFormat() == (m4v ? MediaFormat.M4A : MediaFormat.WEBMA)) {
-                return audio;
+            for (final AudioStream audio : audioStreams) {
+                if (audio.getFormat() == (m4v ? MediaFormat.M4A : MediaFormat.WEBMA)) {
+                    return audio;
+                }
             }
-        }
 
-        if (m4v) {
-            return null;
-        }
+            if (m4v) {
+                return null;
+            }
 
-        // retry, but this time in reverse order
-        for (int i = audioStreams.size() - 1; i >= 0; i--) {
-            final AudioStream audio = audioStreams.get(i);
-            if (audio.getFormat() == MediaFormat.WEBMA_OPUS) {
-                return audio;
+            // retry, but this time in reverse order
```

Comment:
What? Why retry and why in reverse order?

Reply:
This isn't my code, I just reformatted it.

Reply:
thats my comment

Reply:
Yo, kapodamy. Any luck with the multiple sidx boxes?

Reply:
(Please this move this discussion outside this review comment and note the relevant code will not embed to this PR.)